### PR TITLE
Backend improvements (improved function names, better readme, translation improvements, etc.)

### PR DESCRIPTION
**Pull request by Panquesito7**
Things added/changed:

- Improved `README.md` with better explanation and spelling.
- Changed the API keyword from `tp` to `tpr`.
  - Added backward compatibility just in case someone has used our API.
- Added an option to toggle area requests (works kind of tricky as it's disabled by default).
- Added spam prevention for area requests.
- Minor Spanish translation improvements.

## To do

Ready for review.

## How to test

- Ensure everything works as excepted and there are no logical errors or any kind of crash.

### CODE REVIEW

File: functions.lua [hunk 1]
```diff
@@ -503,9 +504,26 @@ function tp.tpc_send(sender, coordinates)
 		if minetest.check_player_privs(pname, {tp_tpc = true}) then
 			local protected = minetest.is_protected(target_coords, sender)
 			if protected then
-				if minetest.get_modpath("areas") then
+				-- Now, we can safely check if the `check_area_requests` option is enabled (see the `config.lua` file, lines 28-35)
+				if minetest.get_modpath("areas") and check_area_requests then
 					for _, area in pairs(areas:getAreasAtPos(target_coords)) do
 						if minetest.get_player_by_name(area.owner) then -- Check if area owners are online
+							-- Spam prevention
+							if spam_prevention[area.owner] == sender and not minetest.check_player_privs(sender, {tp_admin = true}) then
```
Comment (SmallJoker) [inline functions.lua:512]: If there's a similar code snippet somewhere, please offload the check and `minetest.after` call to a helper function.

File: functions.lua [hunk 2]
```diff
@@ -574,116 +592,134 @@ function tp.tpc_send(sender, coordinates)
...
 
-function tp.tpr_deny(name)
-	if not tp.tpr_list[name] and not tp.tphr_list[name]
-	and not tp.tpc_list[name] and not tp.tpn_list[name] then
-		minetest.chat_send_player(name, S("Usage: /tpn allows you to deny teleport/area requests sent to you by other players."))
-		if minetest.get_modpath("chat2") then
-			chat2.send_message(minetest.get_player_by_name(name), S("Usage: /tpn allows you to deny teleport/area requests sent to you by other players."), 0xFFFFFF)
+function tpr.tpr_deny(name)
+	local no_areas_message = "Usage: /tpn allows you to deny teleport requests sent to you by other players."
+	local areas_message = "Usage: /tpn allows you to deny teleport/area requests sent to you by other players."
```
Comment (SmallJoker) [inline functions.lua:597]: `S()` for each.

File: functions.lua [hunk 3]
```diff
@@ -574,116 +592,134 @@ function tp.tpc_send(sender, coordinates)
...
+
+	if not tpr.tpr_list[name] and not tpr.tphr_list[name]
+	and not tpr.tpc_list[name] and not tpr.tpn_list[name] then
+		if check_area_requests then
+			minetest.chat_send_player(name, S(areas_message))
+			if minetest.get_modpath("chat2") then
+				chat2.send_message(minetest.get_player_by_name(name), S(areas_message), 0xFFFFFF)
```
Comment (SmallJoker) [inline functions.lua:604]: `S()` must contain a literal string. The translation parser script can only do static text analysis.

File: functions.lua [hunk 4]
```diff
@@ -574,116 +592,134 @@ function tp.tpc_send(sender, coordinates)
...
+			end
+		else
+			minetest.chat_send_player(name, S(no_areas_message))
+			if minetest.get_modpath("chat2") then
+				chat2.send_message(minetest.get_player_by_name(name), S(no_areas_message), 0xFFFFFF)
```
Comment (SmallJoker) [inline functions.lua:609]: Same as above. This won't be caught by translation parser scripts.

File: functions.lua [hunk 5]
```diff
@@ -574,116 +592,134 @@ function tp.tpc_send(sender, coordinates)
...
-function tp.tpr_accept(name)
-	-- Check to prevent constant teleporting
-	if not tp.tpr_list[name] and not tp.tphr_list[name]
-	and not tp.tpc_list[name] then
-		minetest.chat_send_player(name, S("Usage: /tpy allows you to accept teleport/area requests sent to you by other players."))
-		if minetest.get_modpath("chat2") then
-			chat2.send_message(minetest.get_player_by_name(name), S("Usage: /tpy allows you to accept teleport/area requests sent to you by other players."), 0xFFFFFF)
+function tpr.tpr_accept(name)
+	local no_areas_message = "Usage: /tpy allows you to accept teleport requests sent to you by other players."
+	local areas_message = "Usage: /tpy allows you to accept teleport/area requests sent to you by other players."
```
Comment (SmallJoker) [inline functions.lua:690]: Ditto. Suggestion: (untested)
```Lua
if not tpr.tpr_list[name] and not tpr.tphr_list[name]
and not tpr.tpc_list[name] then
	local msg

	if check_area_requests then
		msg = S("Usage: /tpy allows you to accept teleport requests sent to you by other players.")
	else
		msg = S("Usage: /tpy allows you to accept teleport/area requests sent to you by other players.")
	end
	
	minetest.chat_send_player(name, msg)
	if minetest.get_modpath("chat2") then
		chat2.send_message(minetest.get_player_by_name(name), msg, 0xFFFFFF)
	end
end
```

**Review by SmallJoker — COMMENTED** [review]
I am generally against the renaming due to unnecessary duplication and code changes.
The other tweaks are good ideas, but yet not well implemented.

**Comment by Panquesito7** [comment]
Will be creating another PR to cleanup things after #49 is ready and merged. Thanks.

**Pull request closed without merging**